source/fundamentals/logging.txt

```diff
@@ -0,0 +1,370 @@
+type. The driver provides the following
+built-in specifications for ``LogLevel``:
+
+- ``LevelOff``: suppresses logging.
+- ``LevelInfo``: enables logging of high-level informational messages.
+- ``LevelDebug``: enables logging of detailed debug messages.
+
+You can also specify the severity level by using the enumeration of
+the preceding specifications. The following code shows equivalent ways
+of enabling logging at the ``INFO`` level:
+
+.. code-block:: go
+
+   // Using named value
+   lvl := options.LogLevelInfo
+
+   // Using enumeration
+   lvl := options.LogLevel(1)
+
+.. important::
+
+   The {+driver-short+} currently supports only ``DEBUG`` level messages.
+
+To learn more about logging severity levels, see the Wikipedia entry on
+the :wikipedia:`Syslog standard for message logging <Syslog#Severity_level>`.
+
+Example
+~~~~~~~
+
+This example shows how to configure a basic logger that uses the standard
+logging library and prints log messages. The logger has the following
+specifications:
+
+- The maximum document length is ``25``.
+- The logging component is ``LogComponentCommand``.
+- The logging severity level is ``LogLevelDebug``.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/logging.go
+   :language: go
+   :dedent:
+   :emphasize-lines: 1-4, 9
+   :start-after: start-standard-logger
+   :end-before: end-standard-logger
+
+The following code performs an insert operation, which generates log
+messages:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/fundamentals/code-snippets/logging.go
+      :language: go
+      :dedent:
+      :start-after: start-insert
+      :end-before: end-insert
+
+   .. output::
+      :language: console
+      :visible: false
+
+      {"command":"{\"insert\": \"testColl\",\"or...","commandName":"insert","databaseName":"testDB","driverConnectionId":1,"message":"Command started","operationId":0,"requestId":13,"serverConnectionId":97377,"serverHost":"...","serverPort":27017,"timestamp":...}
+      {"commandName":"insert","driverConnectionId":1,"durationMS":19,"message":"Command succeeded","operationId":0,"reply":"{\"n\": {\"$numberInt\":\"1\"},...","requestId":13,"serverConnectionId":97377,"serverHost":"...","serverPort":27017,"timestamp":...}
+
+.. _golang-logging-custom:
+
+Use a Custom Logger
+-------------------
+
+If the standard logging library does not meet your needs, you can
+implement a custom logger. By customizing your logging configuration,
+you have more control over the content, format, and frequency of log messages.
+
+To use a custom logger, define a logger struct and implement the
+``Info()`` and ``Error()`` methods for the struct. Next, set the logger as
+the ``LogSink`` for your ``Client`` by calling the ``SetSink()`` method on
+your ``LoggerOptions`` instance.
+
+Example
+~~~~~~~
+
+This example demonstrates how to define and implement a custom logger.
+
+1. Define the ``CustomLogger`` struct.
```

Comment:
A colon might tie the instruction to the code sample more closely:

```suggestion
1. Define the ``CustomLogger`` struct:
```

Ditto for other steps